Analytics/Classes/Internal/SEGHTTPClient.m

```diff
@@ -40,13 +41,18 @@ - (NSURLSession *)sessionForWriteKey:(NSString *)writeKey
 {
     NSURLSession *session = self.sessionsByWriteKey[writeKey];
     if (!session) {
-        NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
-        config.HTTPAdditionalHeaders = @{
-            @"Accept-Encoding" : @"gzip",
-            @"Content-Encoding" : @"gzip",
-            @"Content-Type" : @"application/json",
-            @"Authorization" : [@"Basic " stringByAppendingString:[[self class] authorizationHeader:writeKey]],
-        };
+        NSURLSessionConfiguration *config;
+        if (self.configFactory != nil) {
+            config = self.configFactory(writeKey);
+        } else {
+            config = [NSURLSessionConfiguration defaultSessionConfiguration];
+            config.HTTPAdditionalHeaders = @{
+                                             @"Accept-Encoding" : @"gzip",
+                                             @"Content-Encoding" : @"gzip",
+                                             @"Content-Type" : @"application/json",
+                                             @"Authorization" : [@"Basic " stringByAppendingString:[[self class] authorizationHeader:writeKey]],
+                                             };
+        }
```

Comment:
I thought we discussed still let `SEGHTTPClient` add `HTTPAdditionalHeaders` but then subclasses can override and remove the unwanted / un-allowed headers.

Reply:
Right